presto-mysql/src/test/java/io/prestosql/plugin/mysql/TestMySqlTypeMapping.java

```diff
@@ -219,6 +239,214 @@ public void testDecimalExceedingPrecisionMax()
         testUnsupportedDataType("decimal(50,0)");
     }
 
+    @DataProvider
+    public Object[][] testDecimalExceedingPrecisionMaxProvider()
+    {
+        return new Object[][] {
+                {60, 20},
+                {65, 25},
+        };
+    }
+
+    @Test(dataProvider = "testDecimalExceedingPrecisionMaxProvider")
+    public void testDecimalExceedingPrecisionMaxWithUnsupportedValues(Integer typePrecision, Integer typeScale)
+    {
+        JdbcSqlExecutor jdbcSqlExecutor = new JdbcSqlExecutor(mysqlServer.getJdbcUrl());
+
+        List<String> testData = asList(
+                 "1234567890123456789012345678901234567890.1234567890",
+                "-1234567890123456789012345678901234567890.1234567890");
+        int largestScale = testData.stream().map(e -> e.length() - max(e.indexOf("."), 0) - 1).max(Integer::compareTo).orElse(0);
+        int largestIntegerSize = testData.stream().map(e -> (e + ".").replaceFirst("[-]", "").indexOf(".")).max(Integer::compareTo).orElse(0);
+        verify(largestIntegerSize > Decimals.MAX_PRECISION, format("largestIntegerSize must be greater than %d", Decimals.MAX_PRECISION));
+
+        try (TestTable testTable = new TestTable(
+                jdbcSqlExecutor,
+                "tpch.test_exceeding_max_decimal",
+                format("(d_col decimal(%d,%d))", typePrecision, typeScale),
+                testData)) {
+            String testTableName = testTable.getName().split("[.]")[1];
+            for (int scale : asList(0, 16, max(largestScale - 1, 0), largestScale, 38)) {
+                assertQuery(
+                        sessionWithDecimalMappingAllowOverflow(UNNECESSARY, scale),
+                        format("SELECT column_name, data_type FROM information_schema.columns WHERE table_schema = 'tpch' AND table_name = '%s'", testTableName),
+                        format("VALUES ('d_col', 'decimal(%d,%d)')",
+                                Decimals.MAX_PRECISION, calculateDecimalScaleMapping(typePrecision, typeScale, scale)));
```

Comment:
please don't use "product" code to test the "product" code. 

Simply inline the expected values: `"VALUES ('d_col', 'decimal(38,10')"`. This way it is much more readable.